R/bulk_importers.R

```diff
@@ -0,0 +1,61 @@
+
+#' @importFrom R.utils gunzip
+#' @importFrom tools file_path_sans_ext
+#' @importFrom utils download.file
+bulk_importer <- function(db_con, streamable_table){
+  
+  delim <- switch(streamable_table$extension,
+         "tsv" = "\t",
+         "csv" = ",",
+         NULL)
+
+  if(is.null(delim)) return(NULL)
+  
+  quote <- switch(streamable_table$extension,
+                  "tsv" = "",
+                  "csv" = "\"",
+                  NULL)
+  
+  bulk_monetdb <- function(conn, file, tablename){
+    
+    ## if compressed: uncompress first. (remote urls will also be downloaded)
```

Comment:
This whole logic of download-and-uncompress should probably be abstracted out of the monetdb method into something more generic.